src/Pact/Eval.hs

```diff
@@ -394,9 +392,11 @@ loadInterface i@Interface{..} body info gas0 = do
             case dnm of
               Nothing -> return (g, rs)
               Just dn -> do
+                when (isJust $ HM.lookup dn rs) $
+                  evalError (_tInfo t) $ "definition name conflict: " <> pretty dn
                 g' <- computeGas (Left (_tInfo t,dn)) (GModuleMember (MDInterface i))
-                return (g + g',(dn,t):rs)
-      second HM.fromList <$> foldM doDef (gas0,[]) bd
+                return (g + g',HM.insert dn t rs)
+      foldM doDef (gas0,mempty) bd
```

Comment:
On average, a lookup per name in the list of pairs of names and terms is worse complexity than just doing a lookup/insertion via a mad. We also avoid the `fromList` call in addition to any lookups.